core/src/main/java/io/grpc/internal/DnsNameResolver.java

```diff
@@ -95,14 +95,20 @@
   @GuardedBy("this")
   private Listener listener;
 
-  DnsNameResolver(@Nullable String nsAuthority, String name, Attributes params,
+  DnsNameResolver(@Nullable String nsAuthority, URI targetUri, Attributes params,
       Resource<ScheduledExecutorService> timerServiceResource,
       Resource<ExecutorService> executorResource,
-      ProxyDetector proxyDetector) {
+      boolean usingProxy) {
     // TODO: if a DNS server is provided as nsAuthority, use it.
     // https://www.captechconsulting.com/blogs/accessing-the-dusty-corners-of-dns-with-java
     this.timerServiceResource = timerServiceResource;
     this.executorResource = executorResource;
+    this.usingProxy = usingProxy;
+
+    String targetPath = Preconditions.checkNotNull(targetUri.getPath(), "targetPath");
```

Comment:
the code in this constructor (moved some of it in from the `DnsNameResolverProvider` class) is what i extract into `GrpcUtil#unresolvedAddress` but I ran into some test failures while trying to read the host/authority back from the `InetSocketAddress` so right now it's just kinda duplicated in order to get the tests green. just wanted to get some eyes on this change as a whole before I spent the time to remove this duplication